src/Symfony/Component/Messenger/Tests/EnvelopeTest.php

```diff
@@ -114,6 +118,40 @@ public function testWrapWithEnvelope()
         $this->assertCount(1, $envelope->all(DelayStamp::class));
         $this->assertCount(1, $envelope->all(ReceivedStamp::class));
     }
+
+    /**
+     * To be removed in 6.0.
+     *
+     * @group legacy
+     */
+    public function testWithAliases()
+    {
+        $envelope = new Envelope(new \stdClass(), [
+            $s1 = new LegacyDoctrineReceivedStamp(1),
```

Comment:
could you add another test where the new class stamps are added to the envelope but fetching happens with the old aliases.

Reply:
Like this ? https://github.com/symfony/symfony/compare/abc4944bedb3bb27d9f586fd165a10ebc0ff4fa8..6f4e4ca1c4249c76abe97a6423209ae8751c292b

Reply:
it's not so much about `all` but the other tests like 
```
self::assertSame([$s1, $s3], $envelope->all(LegacyDoctrineReceivedStamp::class));
self::assertSame([$s2], $envelope->all(LegacyRedisReceivedStamp::class));
```

Reply:
But `new LegacyDoctrineReceivedStamp(1)` or `new DoctrineReceivedStamp(1)` is strictly the same thing; `LegacyDoctrineReceivedStamp` just being an alias to the `DoctrineReceivedStamp` class, instantiating the envelope with it or with the new fqcn would actually be instantiating the exact same objects. 
Only the alias or fqcn as strings matters here. Instantiating an object with the alias or fqcn doesn't for the php engine.

I'd rather change to instantiating with the new fqcns only if it brings confusion.